WordPress/Classes/ViewRelated/Blog/Blog List/BlogListViewController.m

```diff
@@ -160,6 +168,16 @@ - (void)viewWillAppear:(BOOL)animated
     [self syncBlogs];
     [self setAddSiteBarButtonItem];
     [self updateCurrentBlogSelection];
+    
+    // The logic below should be removed once the `FeatureFlagNewNavBarAppearance` is removed.
+    // There's no need to bypass the blog list once the root VC in "My Site" is the blog details.
+    if (![Feature enabled:FeatureFlagNewNavBarAppearance] && self.canBypassBlogList) {
+        self.canBypassBlogList = false;
```

Comment:
I don't think we want to toggle this to prevent it, if it's possible 🤔  That's a change from the previous behaviour.